### PR TITLE
Support multiple configs in a file

### PR DESCRIPTION
**Pull request by SasSwart**

### CODE REVIEW

File: cmd/can/main.go [hunk 1]
```diff
@@ -1,18 +1,87 @@
...
+
+	absCfgPath, err := filepath.Abs(config.ConfigFilePath)
+	if err != nil {
+		fmt.Printf("could not resolve relative config path: %v\n", err)
+		return
+	}
+	config.ConfigFilePath = absCfgPath
+
+	if config.VersionFlagSet {
+		fmt.Printf("Can: v%s\n", config.SemVer)
```
Comment (alexthemayers) [inline cmd/can/main.go:40]: Let's remove line 19 in favour of this? They're doing the same thing and double printing version info. 
I'm which you think is more favourable, printing at the top of every run and exiting here or printing here and eliminating the top print statement

File: cmd/can/main.go [hunk 2]
```diff
@@ -1,18 +1,87 @@
...
+		os.Exit(0)
+	}
+
+	if config.Debug {
+		fmt.Printf("[v%s]::Using config file \"%s\".\n", config.SemVer, config.ConfigFilePath)
+	}
+
+	configFileReader, err := os.Open(absCfgPath)
+	if err != nil {
+		fmt.Printf("failed to open config file: %v", err)
```
Comment (alexthemayers) [inline cmd/can/main.go:52]: feels like we should os.Exit() here with a non-zero exit code

File: config/config.go [hunk 3]
```diff
@@ -184,28 +156,42 @@ func (d *Data) GetOutputDir() (path string) {
...
 		}
+		d.absOpenAPIPath = filepath.Join(
+			filepath.Dir(ConfigFilePath),
+			d.OpenAPIFile,
+		)
+		return d.absOpenAPIPath
 	}
+
+	if Debug {
+		fmt.Println("Calculating GetOpenAPIFilepath relative to the Executable File location:", d.OpenAPIFile)
```
Comment (alexthemayers) [inline config/config.go:185]: Nice, these log lines will be very useful

File: config/config.go [hunk 4]
```diff
@@ -273,31 +259,31 @@ func (d *Data) resolveTemplateConfig() error {
 	return nil
 }
 
-func (d *Data) setOverridesAndLoadConfig() error {
-	viper.SetConfigFile(ConfigFilePath)
-	err := viper.ReadInConfig()
+func (d *Data) setOverridesAndLoadConfig(reader io.Reader) error {
+	// TODO: Handle this error
```
Comment (SasSwart) [inline config/config.go]: I missed this. Will handle

File: render/go/renderer.go [hunk 5]
```diff
@@ -197,6 +199,8 @@ func ToTitle(s string) (ret string) {
 
 func MustLoadGinServerTestConfig(configPath, openAPIPath string) config.Data {
 	config.ConfigFilePath = configPath
+	absCfgPath, _ := filepath.Abs(config.ConfigFilePath)
```
Comment (alexthemayers) [inline render/go/renderer.go]: Why are we ignoring errors here and in the line below?

File: render/go/renderer.go [hunk 6]
```diff
@@ -205,14 +209,17 @@ func MustLoadGinServerTestConfig(configPath, openAPIPath string) config.Data {
...
-	if err := c.Load(); err != nil {
+	if err := c.Load(configFileReader); err != nil {
 		panic(err)
 	}
 	return c
 }
 func MustLoadGoClientTestConfig(configPath, openAPIPath string) config.Data {
 	// TODO: `Must` this function and load the returned config before returning it
 	config.ConfigFilePath = configPath
+	absCfgPath, _ := filepath.Abs(config.ConfigFilePath)
```
Comment (alexthemayers) [inline render/go/renderer.go]: Errors?

File: cmd/can/main.go (hunk 1 above)
Comment (SasSwart) [reply cmd/can/main.go:40]: Updated

File: cmd/can/main.go (hunk 2 above)
Comment (SasSwart) [reply cmd/can/main.go:52]: done

File: render/go/renderer.go (hunk 6 above)
Comment (SasSwart) [reply render/go/renderer.go]: I was relying on the panic below. These errors would have meant Load can't work. But I've handled them explicitly. Should be more clear now.

File: render/go/renderer.go (hunk 5 above)
Comment (SasSwart) [reply render/go/renderer.go]: I was relying on the panic below. These errors would have meant Load can't work. But I've handled them explicitly. Should be more clear now.

**Review by alexthemayers — APPROVED** [review]

**Pull request merged**